### PR TITLE
Task #165032 chore: Changes in TjCertificate Issued Certificate list view to allow admin to edit Issued Certificates

### PR DESCRIPTION
**Pull request by tushar33**

### CODE REVIEW

File: src/components/com_tjcertificate/administrator/controllers/template.json.php [hunk 1]
```diff
@@ -84,4 +85,27 @@ public function loadDefaultTemplate()
...
+		{
+			echo new JsonResponse(null, Text::_('JINVALID_TOKEN'), true);
+		}
+		else
+		{
+			$app   = Factory::getApplication();
+			$input = $app->input;
+			$templateId = $input->get('templateId');
+			$model = TJCERT::model('Template', array('ignore_request' => true));
+			$templateData = $model->getItem($templateId);
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/template.json.php:106]: use the template library here, and check first if  `$templateId` is empty

File: src/components/com_tjcertificate/administrator/models/certificate.php [hunk 2]
```diff
@@ -23,6 +24,29 @@
  */
 class TjCertificateModelCertificate extends AdminModel
 {
+	/**
+	 * Method to get a template.
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/models/certificate.php:28]: Change from template to certificate

File: src/components/com_tjcertificate/administrator/models/certificate.php [hunk 3]
```diff
@@ -23,6 +24,29 @@
...
+	 *
+	 * @param   integer  $pk  An optional id of the object to get, otherwise the id from the model state is used.
+	 *
+	 * @return  mixed    Template data object on success, false on failure.
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/models/certificate.php:32]: same here

File: src/components/com_tjcertificate/administrator/models/fields/certificatetemplates.php [hunk 4]
```diff
@@ -31,11 +32,17 @@ protected function getOptions()
...
+		$app  = Factory::getApplication()->input;
+		$user = Factory::getUser();
+		$db   = Factory::getDbo();
 
 		$client = $this->getAttribute('client');
 
+		if (empty($client))
+		{
+			$client = $app->get('extension', '');
+		}
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/models/fields/certificatetemplates.php:44]: why do we need this?

File: src/components/com_tjcertificate/media/js/template.js [hunk 5]
```diff
@@ -93,5 +94,30 @@ var template = {
 
 			editor.setValue(templateBody);
 		}
+	},
+	
+	renderCustomTemplate: function(templateId) 
+	{
+		var siteRoot = Joomla.getOptions("system.paths").base;
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/media/js/template.js:101]: use promise here similar to `loadDefaultTemplate`.
add a service to load loadCustomTemplate inside tjCertificateService.js

File: src/components/com_tjcertificate/administrator/controllers/template.json.php (hunk 1 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/template.json.php:106]: done

File: src/components/com_tjcertificate/administrator/models/certificate.php (hunk 2 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/models/certificate.php:28]: done

File: src/components/com_tjcertificate/administrator/models/certificate.php (hunk 3 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/models/certificate.php:32]: done

File: src/components/com_tjcertificate/media/js/template.js (hunk 5 above)
Comment (tushar33) [reply src/components/com_tjcertificate/media/js/template.js]: done

File: src/components/com_tjcertificate/administrator/models/fields/certificatetemplates.php (hunk 4 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/models/fields/certificatetemplates.php:44]: setting client to load templates by client

File: src/components/com_tjcertificate/administrator/controllers/template.json.php [hunk 6]
```diff
@@ -84,4 +85,34 @@ public function loadDefaultTemplate()
 			echo new JResponseJson($templateData);
 		}
 	}
+
+	/**
+	 * Function to load custom template
+	 *
+	 * @return  object  object
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/template.json.php:92]: update return type to object|void

File: src/components/com_tjcertificate/administrator/controllers/template.json.php [hunk 7]
```diff
@@ -84,4 +85,34 @@ public function loadDefaultTemplate()
...
+	 */
+	public function loadCustomTemplate()
+	{
+		if (!JSession::checkToken('get'))
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/template.json.php:96]: use namespace for JSession

File: src/components/com_tjcertificate/administrator/views/certificate/tmpl/edit.php [hunk 8]
```diff
@@ -63,3 +87,41 @@
 	</form>
 </div>
 </div>
+
+<!-- Modal -->
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/views/certificate/tmpl/edit.php:91]: I think this code can be reused, can you create a preview layout and use it in both template & certificate

File: src/components/com_tjcertificate/administrator/controllers/template.json.php (hunk 6 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/template.json.php:92]: done

File: src/components/com_tjcertificate/administrator/controllers/template.json.php (hunk 7 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/controllers/template.json.php:96]: done

File: src/components/com_tjcertificate/administrator/views/certificate/tmpl/edit.php (hunk 8 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/views/certificate/tmpl/edit.php:91]: done

**Pull request merged**